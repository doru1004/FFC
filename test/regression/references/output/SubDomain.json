{
    "form_0": {
        "num_coefficients": 1, 
        "has_cell_integrals": 1, 
        "has_exterior_facet_integrals": 1, 
        "has_interior_facet_integrals": 0, 
        "has_point_integrals": 0, 
        "num_cell_domains": 3, 
        "num_exterior_facet_domains": 6, 
        "num_interior_facet_domains": 0, 
        "num_point_domains": 0, 
        "finite_element_0": {
            "cell_shape": 3, 
            "space_dimension": 4, 
            "value_rank": 0, 
            "evaluate_basis:_0": [0.6513806628275918], 
            "evaluate_basis:_1": [-1.6643198169362], 
            "evaluate_basis:_2": [0.1133045776925702], 
            "evaluate_basis:_3": [1.899634563032329], 
            "evaluate_basis_all": [0.6513806628275918, -1.6643198169362, 0.1133045776925702, 1.899634563032329], 
            "evaluate_basis_derivatives_0_0": [0.6513806628275918], 
            "evaluate_basis_derivatives_0_1": [0.8142105111279443, -2.123895155788242, 0.4873385946136455], 
            "evaluate_basis_derivatives_0_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_1_0": [-1.6643198169362], 
            "evaluate_basis_derivatives_1_1": [0.9133825630703856, 1.145741683084886, 2.192263029714702], 
            "evaluate_basis_derivatives_1_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_2_0": [0.1133045776925702], 
            "evaluate_basis_derivatives_2_1": [-0.04067794577707189, 1.336577910685222, -1.551313955871445], 
            "evaluate_basis_derivatives_2_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_3_0": [1.899634563032329], 
            "evaluate_basis_derivatives_3_1": [-1.686915122451605, -0.3584245005229043, -1.128287592169393], 
            "evaluate_basis_derivatives_3_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_all_0": [0.6513806628275918, -1.6643198169362, 0.1133045776925702, 1.899634563032329], 
            "evaluate_basis_derivatives_all_1": [0.8142105111279443, -2.123895155788242, 0.4873385946136455, 0.9133825630703856, 1.145741683084886, 2.192263029714702, -0.04067794577707189, 1.336577910685222, -1.551313955871445, -1.686915122451605, -0.3584245005229043, -1.128287592169393], 
            "evaluate_basis_derivatives_all_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_dof_0": 0.140720811, 
            "evaluate_dof_1": 0.358429071, 
            "evaluate_dof_2": 0.147608811, 
            "evaluate_dof_3": 0.04602500100000001, 
            "evaluate_dofs": [0.140720811, 0.358429071, 0.147608811, 0.04602500100000001], 
            "interpolate_vertex_values": [0.140720811, 0.358429071, 0.147608811, 0.04602500100000001], 
            "num_sub_elements": 0, 
        },
        "dofmap_0": {
            "needs_mesh_entities_0": 1, 
            "needs_mesh_entities_1": 0, 
            "needs_mesh_entities_2": 0, 
            "needs_mesh_entities_3": 0, 
            "global_dimension": 10001, 
            "local_dimension": 4, 
            "max_local_dimension": 4, 
            "geometric_dimension": 3, 
            "num_facet_dofs": 3, 
            "num_entity_dofs_0": 1, 
            "num_entity_dofs_1": 0, 
            "num_entity_dofs_2": 0, 
            "num_entity_dofs_3": 0, 
            "tabulate_dofs": [0, 0, 0, 0], 
            "tabulate_facet_dofs_0": [1, 2, 3], 
            "tabulate_facet_dofs_1": [0, 2, 3], 
            "tabulate_facet_dofs_2": [0, 1, 3], 
            "tabulate_facet_dofs_3": [0, 1, 2], 
            "tabulate_entity_dofs_0_0": [0], 
            "tabulate_entity_dofs_0_1": [1], 
            "tabulate_entity_dofs_0_2": [2], 
            "tabulate_entity_dofs_0_3": [3], 
            "tabulate_entity_dofs_1_0": [], 
            "tabulate_entity_dofs_1_1": [], 
            "tabulate_entity_dofs_1_2": [], 
            "tabulate_entity_dofs_1_3": [], 
            "tabulate_entity_dofs_1_4": [], 
            "tabulate_entity_dofs_1_5": [], 
            "tabulate_entity_dofs_2_0": [], 
            "tabulate_entity_dofs_2_1": [], 
            "tabulate_entity_dofs_2_2": [], 
            "tabulate_entity_dofs_2_3": [], 
            "tabulate_entity_dofs_3_0": [], 
            "tabulate_coordinates_0": [0.903, 0.341, 0.457], 
            "tabulate_coordinates_1": [0.5610000000000001, 0.767, 0.833], 
            "tabulate_coordinates_2": [0.987, 0.783, 0.191], 
            "tabulate_coordinates_3": [0.123, 0.5610000000000001, 0.667], 
            "num_sub_dofmaps": 0, 
        },
        "default_cell_integral": 0, 
        "cell_integral_2": {
            "tabulate_tensor": [0.0069038250552306], 
        },
        "default_exterior_facet_integral": 0, 
        "exterior_facet_integral_5": {
            "tabulate_tensor_0": [0.0578157034656348], 
            "tabulate_tensor_1": [0.05825405301080629], 
            "tabulate_tensor_2": [0.03959102291945829], 
            "tabulate_tensor_3": [0.03414692020900983], 
        },
        "default_interior_facet_integral": 0, 
        "default_point_integral": 0, 
    },
}
