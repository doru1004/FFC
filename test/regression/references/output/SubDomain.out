
Testing form
----------------------
0_num_coefficients = 1
1_has_cell_integrals = 1
2_has_exterior_facet_integrals = 1
3_has_interior_facet_integrals = 0
4_has_point_integrals = 0
5_num_cell_domains = 3
6_num_exterior_facet_domains = 6
7_num_interior_facet_domains = 0
8_num_point_domains = 0

Testing finite_element
----------------------
9_cell_shape = 3
10_space_dimension = 4
11_value_rank = 0
12_evaluate_basis:_0 = 0.6513806628275918
13_evaluate_basis:_1 = -1.6643198169362
14_evaluate_basis:_2 = 0.1133045776925702
15_evaluate_basis:_3 = 1.899634563032329
16_evaluate_basis_all = 0.6513806628275918 -1.6643198169362 0.1133045776925702 1.899634563032329
17_evaluate_basis_derivatives_0_0 = 0.6513806628275918
18_evaluate_basis_derivatives_0_1 = 0.8142105111279443 -2.123895155788242 0.4873385946136455
19_evaluate_basis_derivatives_0_2 = 0 0 0 0 0 0 0 0 0
20_evaluate_basis_derivatives_1_0 = -1.6643198169362
21_evaluate_basis_derivatives_1_1 = 0.9133825630703856 1.145741683084886 2.192263029714702
22_evaluate_basis_derivatives_1_2 = 0 0 0 0 0 0 0 0 0
23_evaluate_basis_derivatives_2_0 = 0.1133045776925702
24_evaluate_basis_derivatives_2_1 = -0.04067794577707189 1.336577910685222 -1.551313955871445
25_evaluate_basis_derivatives_2_2 = 0 0 0 0 0 0 0 0 0
26_evaluate_basis_derivatives_3_0 = 1.899634563032329
27_evaluate_basis_derivatives_3_1 = -1.686915122451605 -0.3584245005229043 -1.128287592169393
28_evaluate_basis_derivatives_3_2 = 0 0 0 0 0 0 0 0 0
29_evaluate_basis_derivatives_all_0 = 0.6513806628275918 -1.6643198169362 0.1133045776925702 1.899634563032329
30_evaluate_basis_derivatives_all_1 = 0.8142105111279443 -2.123895155788242 0.4873385946136455 0.9133825630703856 1.145741683084886 2.192263029714702 -0.04067794577707189 1.336577910685222 -1.551313955871445 -1.686915122451605 -0.3584245005229043 -1.128287592169393
31_evaluate_basis_derivatives_all_2 = 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
32_evaluate_dof_0 = 0.140720811
33_evaluate_dof_1 = 0.358429071
34_evaluate_dof_2 = 0.147608811
35_evaluate_dof_3 = 0.04602500100000001
36_evaluate_dofs = 0.140720811 0.358429071 0.147608811 0.04602500100000001
37_interpolate_vertex_values = 0.140720811 0.358429071 0.147608811 0.04602500100000001
38_num_sub_elements = 0

Testing dofmap
----------------------
39_needs_mesh_entities_0 = 1
40_needs_mesh_entities_1 = 0
41_needs_mesh_entities_2 = 0
42_needs_mesh_entities_3 = 0
43_global_dimension = 10001
44_local_dimension = 4
45_max_local_dimension = 4
46_geometric_dimension = 3
47_num_facet_dofs = 3
48_num_entity_dofs_0 = 1
49_num_entity_dofs_1 = 0
50_num_entity_dofs_2 = 0
51_num_entity_dofs_3 = 0
52_tabulate_dofs = 0 0 0 0
53_tabulate_facet_dofs_0 = 1 2 3
54_tabulate_facet_dofs_1 = 0 2 3
55_tabulate_facet_dofs_2 = 0 1 3
56_tabulate_facet_dofs_3 = 0 1 2
57_tabulate_entity_dofs_0_0 = 0
58_tabulate_entity_dofs_0_1 = 1
59_tabulate_entity_dofs_0_2 = 2
60_tabulate_entity_dofs_0_3 = 3
61_tabulate_entity_dofs_1_0 =
62_tabulate_entity_dofs_1_1 =
63_tabulate_entity_dofs_1_2 =
64_tabulate_entity_dofs_1_3 =
65_tabulate_entity_dofs_1_4 =
66_tabulate_entity_dofs_1_5 =
67_tabulate_entity_dofs_2_0 =
68_tabulate_entity_dofs_2_1 =
69_tabulate_entity_dofs_2_2 =
70_tabulate_entity_dofs_2_3 =
71_tabulate_entity_dofs_3_0 =
72_tabulate_coordinates_0 = 0.903 0.341 0.457
73_tabulate_coordinates_1 = 0.5610000000000001 0.767 0.833
74_tabulate_coordinates_2 = 0.987 0.783 0.191
75_tabulate_coordinates_3 = 0.123 0.5610000000000001 0.667
76_num_sub_dofmaps = 0
77_default_cell_integral = 0

Testing cell_integral
----------------------
78_tabulate_tensor = 0.0069038250552306
79_default_exterior_facet_integral = 0

Testing exterior_facet_integral
----------------------
80_tabulate_tensor_0 = 0.0578157034656348
81_tabulate_tensor_1 = 0.05825405301080629
82_tabulate_tensor_2 = 0.03959102291945829
83_tabulate_tensor_3 = 0.03414692020900983
84_default_interior_facet_integral = 0
85_default_point_integral = 0
